watertap/examples/flowsheets/oaro/oaro_multi.py

```diff
@@ -0,0 +1,1423 @@
+
+    # flowsheet set up
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(dynamic=False)
+    m.fs.erd_type = erd_type
+    m.fs.properties = props.NaClParameterBlock()
+    m.fs.costing = WaterTAPCosting()
+
+    # stage set up
+    m.fs.NumberOfStages = Param(initialize=number_of_stages)
+    m.fs.Stages = RangeSet(m.fs.NumberOfStages)
+    m.fs.NonFirstStages = RangeSet(2, m.fs.NumberOfStages)
+    m.fs.NonFinalStages = RangeSet(m.fs.NumberOfStages - 1)
+    if number_of_stages > 2:
+        m.fs.IntermediateStages = RangeSet(2, m.fs.NumberOfStages - 1)
+    else:
+        m.fs.IntermediateStages = RangeSet(0)
+    m.fs.FirstStage = m.fs.Stages.first()
+    m.fs.LastStage = m.fs.Stages.last()
+
+    # Control volume flow blocks
+    m.fs.feed = Feed(property_package=m.fs.properties)
+    m.fs.product = Product(property_package=m.fs.properties)
+    m.fs.disposal = Product(property_package=m.fs.properties)
+
+    # --- Main pump ---
+    m.fs.PrimaryPumps = Pump(m.fs.Stages, property_package=m.fs.properties)
+    for pump in m.fs.PrimaryPumps.values():
+        pump.costing = UnitModelCostingBlock(
+            flowsheet_costing_block=m.fs.costing,
+        )
+
+    # --- Recycle pump ---
+    m.fs.RecyclePumps = Pump(m.fs.NonFirstStages, property_package=m.fs.properties)
+    for pump in m.fs.RecyclePumps.values():
+        pump.costing = UnitModelCostingBlock(
+            flowsheet_costing_block=m.fs.costing, costing_method=cost_low_pressure_pump
+        )
+
+    m.fs.total_pump_work = Expression(
+        expr=sum(
+            pyunits.convert(pump.work_mechanical[0], to_units=pyunits.kW)
+            for pump in itertools.chain(
+                m.fs.PrimaryPumps.values(), m.fs.RecyclePumps.values()
+            )
+        )
+    )
+
+    # --- Reverse Osmosis Block ---
+    m.fs.RO = ReverseOsmosis0D(
+        property_package=m.fs.properties,
+        has_pressure_change=True,
+        pressure_change_type=PressureChangeType.calculated,
+        mass_transfer_coefficient=MassTransferCoefficient.calculated,
+        concentration_polarization_type=ConcentrationPolarizationType.calculated,
+        has_full_reporting=True,
+    )
+    m.fs.RO.costing = UnitModelCostingBlock(flowsheet_costing_block=m.fs.costing)
+
+    # --- Osmotically Assisted Reverse Osmosis Block ---
+    m.fs.OAROUnits = OsmoticallyAssistedReverseOsmosis0D(
+        m.fs.NonFinalStages,
+        property_package=m.fs.properties,
+        has_pressure_change=True,
+        pressure_change_type=PressureChangeType.calculated,
+        mass_transfer_coefficient=MassTransferCoefficient.calculated,
+        concentration_polarization_type=ConcentrationPolarizationType.calculated,
+        has_full_reporting=True,
+    )
+    for stage in m.fs.OAROUnits.values():
+        stage.costing = UnitModelCostingBlock(
+            flowsheet_costing_block=m.fs.costing,
+            costing_method_arguments={"oaro_type": "high_pressure"},
```

Comment:
no need for distinguishing between "high_pressure" and "standard" like we do for RO. Can be revised in future PR.